sympy/core/numbers.py

```diff
@@ -1308,73 +1329,96 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self == other
 
-    def __gt__(self, other):
+    def _Frel(self, other, op):
+        from sympy.core.evalf import evalf
+        from sympy.core.numbers import prec_to_dps
         try:
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s > %s" % (self, other))
-        if other.is_NumberSymbol:
-            return other.__lt__(self)
-        if other.is_Rational and not other.is_Integer:
-            self *= other.q
-            other = _sympify(other.p)
-        elif other.is_comparable:
-            other = other.evalf()
-        if other.is_Number and other is not S.NaN:
+        if other.is_Rational:
+            from sympy.functions.elementary.complexes import sign
+            from sympy.functions.elementary.exponential import log
+            # we don't want to lose precision in other when making
+            # the comparison and we don't want to express Float
+            # as a Rational because it has the potential to create
+            # very large Rationals, so we try identify easy cases
+            # and finally use logarithms to make the determination
+            if sign(self) != sign(other):
+                s = Float._new([fnone, fzero, fone][sign(self) + 1], 1, zero=False)
+                o = Float._new([fnone, fzero, fone][sign(other) + 1], 1, zero=False)
+            else:
+                s = self
+                o = Float._new(other._as_mpf_val(self._prec), self._prec, zero=False)
```

Comment:
I feel uneasy about this conversion. It may lead to equal mpfs for unequal numbers.
```
>>> a = Float('2/3')  # self
>>> b = Rational(2, 3)  # other
>>> a._mpf_
(0, mpz(6004799503160661), -53L, 53L)
>>> b._as_mpf_val(a._prec)
(0, mpz(6004799503160661), -53L, 53L)
```

Reply:
Yes, that is why the check is made in the next line to see whether `Rational(o) != other`. If so, it means there was a loss in the conversion and the more accurate test will be made, otherwise the conversion to Float was lossless.

Reply:
```python
>>> a = Float('2/3')
>>> b=Rational(2,3)
>>> Rational(Float._new(b._as_mpf_val(a._prec), a._prec))
6004799503160661/9007199254740992
>>> Rational(Float._new(b._as_mpf_val(a._prec), a._prec))==b
False
```

Reply:
I would expect `Rational(o) != other` for most rational numbers. Could we just pass to the more accurate test?

Reply:
OK, change has been made.